src/PortMapping.php

```diff
@@ -17,6 +17,20 @@ public function __construct(int $portOnHost, int $portOnDocker)
 
     public function __toString()
     {
-        return "-p {$this->portOnHost}:{$this->portOnDocker}";
+        if ( -1 == $this->portOnHost || -1 == $this->portOnDocker ) {
+            return "-P ";
+        } else {
```

Comment:
An  `else` isn't needed as you're already returning early.

Reply:
Yeah... I just did it for symmetry, but I can remove it

Reply:
An extra space crept into the "-P" and broke the test. I fixed that too.